manual/api/vector_search.md

```diff
@@ -0,0 +1,221 @@
+## Vecoter Search
+We have developed vector search functionality as an extension for SeaSearch. The following introduces the related APIs.
+
+### Creating a Vector Index
+To use the vector search feature, you must create a vector index. This can be done via mapping.
+
+For example, we create an index and set the vector field in the document data to be written as "vec". The index type is set to `flat` and the vector dimension is `768`.
+
+```
+[PUT] /es/${indexName}/_mapping
+
+body:
+{
+  "properties": {
+    "vec": {
+      "type": "vector",
+      "dims": 768,
+      "vec_index_type": "flat"
+    }
+  }
+}
+```
+
+#### Parameter Description:
+- ${indexName}: The name of the index.
+- type: Fixed as `vector`, indicating that this is a vector index.
+- dims: The vector dimension.
+- m: A parameter required for the `ivf_pq` index, it must be divisible by dims. For example, the `dims` is `768`, the `m` could be `192`.
+- nbits: A parameter required for the `ivf_pq` index, must grater than 0, we recommend `4` or `8`.
+- vec_index_type: The index type. Supported types are `flat` and `ivf_pq`.
+
+### Indexing Documents Containing Vectors
+
+Indexing a document that includes vectors is identical at the API level to indexing a regular document. 
+You may choose the method that best suits your needs. The following example uses the bulk API.
+
+```
+[POST] /es/_bulk
+
+{ "index" : { "_index" : "index1" } }
+{"name": "jack1", "vec": [10.2, 10.41, 9.5, 22.2]}
+{ "index" : { "_index" : "index1" } }
+{"name": "jack2", "vec": [10.2, 11.41, 9.5, 22.2]}
+{ "index" : { "_index" : "index1" } }
+{"name": "jack3", "vec": [10.2, 12.41, 9.5, 22.2]}
+```
+
+### Searching Vectors
+
+You can search the system for the top N similar vectors and return the corresponding document information.
```

Comment:
You can search the index for the top K similar vectors to the input vector and return information from the documents that contain the vectors.